src/compiler/scala/tools/reflect/ReflectGlobal.scala

```diff
@@ -11,6 +12,12 @@ import scala.tools.nsc.Settings
 class ReflectGlobal(currentSettings: Settings, reporter: Reporter, override val rootClassLoader: ClassLoader)
   extends Global(currentSettings, reporter) with scala.tools.reflect.ReflectSetup with scala.reflect.runtime.SymbolTable {
 
+  override lazy val analyzer = new {
+    val global: ReflectGlobal.this.type = ReflectGlobal.this
+  } with Analyzer {
+    override lazy val macroParentClassLoader = rootClassLoader
+  }
+
```

Comment:
I think I'd prefer a smaller API surface, especially given we already have a precedent that we'd have to change otherwise (risks). Also, having findMacroClassLoader as an ultimate extension point means more flexibility, because then we're not confined to always creating a classloader from the compiler's classpath + something else.